### PR TITLE
Fix forbidden access page in Front Office

### LINKED ISSUE #27094: Error in Front controller when checkAccess() returns false
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

While documenting the flow of Front controllers, I found a rather peculiar call path:

Weird call path

As of 1.7.8, Front controllers [systematically call `displayMaintenancePage()`](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/FrontController.php#L493). This method will display the maintenance page and exit [only if the maintenance mode is activated](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/FrontController.php#L763). Otherwise, it will let the flow of execution continue.

The execution flow (standard for front and back) will then call the `checkAccess()` method, theoretically [to determine if the current user can access the page or not](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/Controller.php#L155). In admin controllers, this will check the XSRF token. In Front controllers, this [returns returns true by default](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/FrontController.php#L247) (except in AuthController, where it [will redirect if the user is already logged in](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/FrontController.php#L247)).

The standard behavior of controllers is to call `initCursedPage()` when `checkAccess()` returns false. In Admin controllers, this will [display the "invalid token" page](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/AdminController.php#L2288) that we all know and love. In Front controllers, oddly, this [redirects the flow to  `displayMaintenancePage()`](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/FrontController.php#L611) again, although at this point in we already know that the site is not in maintenance mode. This means that execution flow will always continue, leading to this error:

Error

For all we know, this bug might be at least 10 years old, and it only happens when `checkAccess()` returns false **and** the shop is not in maintenance mode.

### Expected behavior

I expected to get an "access forbidden" error message.

### Steps to reproduce

This can be reproduced easily by modifying `FrontController::checkAccess()` to make it return false instead of true (like a custom controller would do). After having modified this, access any Front office URL and you will get the error.

### PrestaShop version(s) where the bug happened

I think this affects all versions since at least 1.5

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | There is a logic error in Front controllers described in the original ticket. This change fixes the bug by using the already existing, but unused, "forbidden" template.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27094
| How to test?      | See issue to reproduce. With the PR, you get a nice error message instead of a crash.
| Possible impacts? | This only affects access to the Front office. If you put the shop in maintenance mode, you should still get the maintenance page, even if checkAccess() returns false.

## Details

Make sure to read the issue to understand the original bug.

I made `initCursedPage()` show an error page instead of redirecting to `displayMaintenancePage()` which will never display the maintenance page, since the shop has to not be in maintenance mode to get to this method being called.

The "forbidden" error template existed already, but wasn't used anywhere. I updated it to look like the "restricted country" template (the one you get when the shop is not accessible from your country), an a simple wording:

Forbidden page

I searched and I found that initCursedPage() is only called (by the Core at least) in the explained cases. So there are no caveats as far as I can tell.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27095)

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]
LGTM

I also tested the behavior with and without the PR, it's working as described in the PR 👍 

(waiting for wording before we can merge)

**Review by Julievrz — APPROVED** [review]
Wording ✅

**Comment by Robin-Fischer-PS** [comment]
OK for QA by dev :heavy_check_mark: !

**Comment by eternoendless** [comment]
Thanks everyone!

**Pull request merged**
